DependencyInjection/Compiler/ResourcePass.php

```diff
@@ -37,24 +37,40 @@ public function process(ContainerBuilder $container)
         foreach ($container->findTaggedServiceIds('api.resource') as $serviceId => $tags) {
             $resourceReferences[] = new Reference($serviceId);
             $resourceDefinition = $container->getDefinition($serviceId);
+            $serviceProperties = $resourceDefinition->getProperties();
 
             if (self::BUILTIN_RESOURCE !== $this->getClass($container, $resourceDefinition)) {
                 continue;
             }
 
+            // get the properties and reset them to avoid property-injection
+            $serviceProperties = $resourceDefinition->getProperties();
+            $resourceDefinition->setProperties([]);
+
             if (!$resourceDefinition->hasMethodCall('initItemOperations')) {
-                $resourceDefinition->addMethodCall('initItemOperations', [[
-                    $this->createOperation($container, $serviceId, 'GET', false),
-                    $this->createOperation($container, $serviceId, 'PUT', false),
-                    $this->createOperation($container, $serviceId, 'DELETE', false),
-                ]]);
+                if (empty($serviceProperties['itemOperations'])) {
```

Comment:
Can you move this bit of code into a method for both items and collections to reduce duplication ?